src/Microsoft.NetCore.Analyzers/Core/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1074,4 +1074,22 @@
   <data name="JsonNetMaybeInsecureSerializerTitle" xml:space="preserve">
     <value>Ensure that JsonSerializer has a secure configuration when deserializing</value>
   </data>
+  <data name="UseDefaultDllImportSearchPathsAttribute" xml:space="preserve">
+    <value>Use DefaultDllImportSearchPaths attribute for P/Invokes</value>
+  </data>
+  <data name="UseDefaultDllImportSearchPathsAttributeDescription" xml:space="preserve">
+    <value>By default, P/Invokes using DllImportAttribute probe a number of directories, including the current working directory for the library to load. This can be a security issue for certain applications, leading to DLL hijacking.</value>
+  </data>
+  <data name="UseDefaultDllImportSearchPathsAttributeMessage" xml:space="preserve">
+    <value>The method {0} didn't use DefaultDllImportSearchPaths attribute for P/Invokes.</value>
+  </data>
+  <data name="DoNotUseUnsafeDllImportSearchPath" xml:space="preserve">
+    <value>Do not use unsafe DllImportSearchPath value</value>
+  </data>
+  <data name="DoNotUseUnsafeDllImportSearchPathDescription" xml:space="preserve">
+    <value>There could be malicious DLL under the application directory or the default DLL search directories. Use an DllImportSearchPath value that sepecifies explicit search path instead</value>
```

Comment:
>There could be malicious DLL under the application directory or the default DLL search directories. Use an DllImportSearchPath value that sepecifies explicit search path instead [](start = 11, length = 177)

The could be a malicious DLL in the default DLL search directories. Or, depending on where your application is run from, there could be a malicious DLL in the application's directory. Use a DllImportSearchPath value that specifies an explicit search path instead. The DllImportSearchPath flags that this rule looks for can be configured in .editorconfig.